### ISSUE TITLE
failing to test code

### ISSUE THREAD

**Issue by chrisclarkson** [brielin/bimmer#21]
Hi I am very interested in getting your code to work with my own data. 
However I am struggling to test the code...
I installed bimmer and inspre with:
```
devtools::install_gihub('https://github.com/brielin/bimmer/')
devtools::install_gihub('https://github.com/brielin/inspre/')
```
After a lot of fiddling they both installed:

I run `Rscript testthat.R` and get  the following:
 ```Rscript testthat.R
‚îÄ‚îÄ 1. Error: (unknown) (@test-fit-model.R#1)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
:-(
Backtrace:
 1. bimmer:::generate_dataset(...)

‚îÄ‚îÄ 2. Error: get_direct_observed_works (@test-simulate.R#5)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
:-(
Backtrace:
 1. bimmer:::generate_dataset(...)

‚îÄ‚îÄ 3. Error: generate_network_runs (@test-simulate.R#33)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
unused argument (normalize = FALSE)

‚îÄ‚îÄ 4. Error: generate_network_normalizes (@test-simulate.R#40)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
could not find function "huge.generator"
Backtrace:
 1. bimmer:::generate_network(D, p)

‚îÄ‚îÄ 5. Error: generate_dataset_runs (@test-simulate.R#46)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
:-(
Backtrace:
 1. bimmer:::generate_dataset(...)

‚îÄ‚îÄ 6. Error: select_snps_oracle_works (@test-simulate.R#59)  ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ
:-(
Backtrace:
 1. bimmer:::generate_dataset(...)

‚ïê‚ïê testthat results  ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê
[ OK: 3 | SKIPPED: 0 | WARNINGS: 0 | FAILED: 6 ]
1. Error: (unknown) (@test-fit-model.R#1)
2. Error: get_direct_observed_works (@test-simulate.R#5)
3. Error: generate_network_runs (@test-simulate.R#33)
4. Error: generate_network_normalizes (@test-simulate.R#40)
5. Error: generate_dataset_runs (@test-simulate.R#46)
6. Error: select_snps_oracle_works (@test-simulate.R#59)
```

I try loading the library in R and using a function:

```
 library(bimmer)
 bimmer::dataset <- generate_dataset(N = 100, D = 3, M_total = 20, M_s = 5, M_p = 5, prop_shared = NULL,rho = 0, noise = 0.5, p_net = 1, sd_net = 0.1)

 Error in generate_dataset(N = 100, D = 3, M_total = 20, M_s = 5, M_p = 5,  :
   could not find function "generate_dataset"
```

**Comment by brielin** [comment]
Hi Chris,

This is brand new software and I'm the only one that has used it so far, so please be patient while I work through the kinks to enable others to use it as well. I appreciate that you're trying to work with it as that will help me through the process!

One thing to note is that most of the simulation code isn't exported, that is probably why you're seeing the `could not find function` message. Try running `devtools::load_all()` first to load the functions that aren't exported, then let me know what happens. I plan to try running through this from scratch and fleshing out the user manual more this week and I will update you when I push new builds.

Thanks again!
Brielin

**Comment by chrisclarkson** [comment]
Hi Brielin,
Thanks so much for getting back to me so quickly! No problem- I totally understand. Congrats on the really cool paper!
PS. I tried `devtools::load_all()` in an R session and then to generate a dataset but got the same error.

I tried also inserting `load_all` into the `testthat.R` script and running it but got all the same errors.

**Comment by brielin** [comment]
Hi Chris,

Thanks for your patience. I made some edits to the tests and added some new instructions to the README.md regarding the proper way to install and run them.

I noticed in your original message you said you got them installed "after a lot of fiddling". I am very curious what fiddling was required? I was able to install this on another machine easily just using devtools::install_github(). If there are other issues you ran into I'd like to take care of them as well.

Cheers,
Brielin

**Comment by chrisclarkson** [comment]
Hi Brielin,
I re-installed the package as shown on your instructions.
It gave an error while updating the packages saying the igraph is not available for R-3.6.... 
But when I load the library and run the tests all 60 tests run.
I will let you know if I have any other issues.
Thanks for your help.
:)
